core/src/main/java/io/substrait/extension/SimpleExtension.java

```diff
@@ -796,20 +804,43 @@ public static ExtensionCollection load(String namespace, InputStream stream) {
 
   public static ExtensionCollection buildExtensionCollection(
       String namespace, ExtensionSignatures extensionSignatures) {
+    List<ScalarFunctionVariant> scalarFunctionVariants =
+        extensionSignatures.scalars().stream()
+            .flatMap(t -> t.resolve(namespace))
+            .collect(java.util.stream.Collectors.toList());
+
+    List<AggregateFunctionVariant> aggregateFunctionVariants =
+        extensionSignatures.aggregates().stream()
+            .flatMap(t -> t.resolve(namespace))
+            .collect(java.util.stream.Collectors.toList());
+
+    Stream<WindowFunctionVariant> windowFunctionVariants =
+        extensionSignatures.windows().stream().flatMap(t -> t.resolve(namespace));
+
+    // Aggregate functions can be used as Window Functions
+    Stream<WindowFunctionVariant> windowAggFunctionVariants =
+        aggregateFunctionVariants.stream()
+            .map(
+                afi ->
+                    WindowFunctionVariant.builder()
+                        // Sets all fields declared in the Function interface
+                        .from(afi)
+                        // Set WindowFunctionVariant fields
+                        .decomposability(afi.decomposability())
+                        .intermediate(afi.intermediate())
+                        // Aggregate Functions used in Windows have WindowType Streaming
+                        .windowType(SimpleExtension.WindowType.STREAMING)
+                        .build());
+
+    List<WindowFunctionVariant> allWindowFunctionVariants =
+        Stream.concat(windowFunctionVariants, windowAggFunctionVariants)
+            .collect(Collectors.toList());
```

Comment:
The spec is being loosed to allow usage of aggregate functions as window function.
https://github.com/substrait-io/substrait/pull/540

`substrait-java` already supported this behaviour, but the implementation is simpler/cleaner if aggregate functions are added as valid window functions here.